packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -356,6 +358,7 @@ contract OptimismPortal is Initializable, ResourceMetering, OnApprove, ISemver {
         bool transferStatus = true;
         if (_tx.value > 0) {
             transferStatus = IERC20(nativeTokenAddress).transfer(_tx.target, _tx.value);
```

Comment:
@nguyenzung 
I'm sorry. I missed it. Do you think we have to change `transfer` -> `approve` ?

Reply:
I think we should keep it. The reason we use approve in another function: it is possible for relay functions with external contract addresses. In finalizeWithdrawalTransaction(), the target is always CrossDomainMessenger:relayMessage .

Reply:
> I think it is possible the target is external contract addresses in case of withdrawal as well (There is no limit on data in initiateWithdrawal in L2ToL1MessagePasser.sol). If there are no functional or security issues, what�do you think about using approve?

Please correct me if I am wrong!

If there is an external targets, it will be failed when proveWithdrawalTransaction() is called because when trigger a withdraw in L2, L2ToL1MessagePasser:initiateWithdrawal will record a withdrawalHash with the target is relayMessage for proving in proveWithdrawalTransaction(). 

I think we can also use approve(), but there is more gas cost and we don't really need it.

Reply:
Oh, you are right, anyone can call initiateWithdrawal()
let me update to approve()

Reply:
Thank you very much, Steven

Reply:
Thank you for you detail check. I sincerely apologize for the many chaotic reviews and confusion.
Thank you so much for your code.

Reply:
Never mind, Steven
The review has been really helpful to me

Reply:
@rlgns98kr In case of using approve(), is it okay if I increase the depositedAmount? Could you check it again

Reply:
Great point! I think it is okay.
We want to use `depositedAmount` for our staking. So it can be different with TON balance of `Portal`. And I think it is correct to increase/decrease when there is approve.
@boohyung @nguyenzung 
Do we need to check Contract team about it?

Reply:
I agree with you. I think when we approve, basically, the amount token can be taken at anytime. 
In my opinion. It is good for having their support. I think we still have anther audit process, is it right?

Reply:
Thanks. @rlgns98kr @nguyenzung 
I think it's reasonable for us. but we need further checker our bridge contracts for avoid unexpected issue. So let me ask about the additional internal audit with Contract team this week.

If they say yes, we should prepare below.
* bridge contracts
* unit test code
* example code (if we can prepare. it is optional)